### PR TITLE
Commit for Action and Func delegates

### PR DESCRIPTION
**Pull request by namcogit**
Please check and verify my code process and update if anything needed to be changed accordingly.

### CODE REVIEW

File: csharp-advanced-topics/ActionInCSharp/ActionInCSharp/Program.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+
+namespace ActionInCSharp
+{
+    class Program
+    {
+        private static int resultNumber = 0;
+        private static string resultMessage = "";
+        static void Main(string[] args)
```
Comment (CodeMazeBlog) [inline csharp-advanced-topics/ActionInCSharp/ActionInCSharp/Program.cs:9]: Newline needed

File: csharp-advanced-topics/ActionInCSharp/ActionInCSharp/ActionInCSharp.csproj [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (CodeMazeBlog) [inline csharp-advanced-topics/ActionInCSharp/ActionInCSharp/ActionInCSharp.csproj:5]: We use .NET 6 for all our articles right now. You might need to install VS2022

File: csharp-advanced-topics/ActionInCSharp/TestScenario/UnitTestAction.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+namespace TestScenario
+{
+    [TestClass]
+    public class UnitTestAction
+    {
+        private static int resultNumber = 0;
+        private static string resultMessage = "";
+        [TestMethod]
+        public void When_PassingSingleValue()
+        {
```
Comment (CodeMazeBlog) [inline csharp-advanced-topics/ActionInCSharp/TestScenario/UnitTestAction.cs:13]: Test names are almost what they should be. When part should be more descriptive, and you need a then part. Please check the guidelines for proper test naming.

File: csharp-advanced-topics/ActionInCSharp/TestScenario/UnitTestAction.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            MultiplicationMethod(5, 3);
+
+            //Number match
+            Assert.AreEqual(resultNumber, 15);
+
+            //Number mis-match
+            Assert.AreNotEqual(resultNumber, 10);
+
+            //Condition check
+            Assert.IsTrue(resultNumber > 0);
```
Comment (CodeMazeBlog) [inline csharp-advanced-topics/ActionInCSharp/TestScenario/UnitTestAction.cs:37]: Comments are not needed, Asserts are descriptive enough.

File: csharp-advanced-topics/FuncInCSharp/FuncInCSharp/FuncInCSharp.csproj [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
```
Comment (CodeMazeBlog) [inline csharp-advanced-topics/FuncInCSharp/FuncInCSharp/FuncInCSharp.csproj:6]: .NET 6

File: csharp-advanced-topics/FuncInCSharp/TestScenario/UnitTestAction.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            //Pass single input parameter
+            Func<int, double> SingleFunc = SingleFuncNumber;
+            var val = SingleFunc(5);
+
+            //Length match
+            Assert.AreNotEqual(val, 2200);
+
+            //Condition check
+            Assert.IsTrue(val > 0);
+        }
```
Comment (CodeMazeBlog) [inline csharp-advanced-topics/FuncInCSharp/TestScenario/UnitTestAction.cs:21]: Naming fix, and comments should be removed.

File: csharp-advanced-topics/FuncInCSharp/FuncInCSharp/Program.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        private static string resultMessage;
+        static void Main(string[] args)
+        {
+            //without lambda resultMessage = MutipleFuncMethod(5, 3);
+            Func<int, int, string> MutipleFuncMethod = MultipleFuncNumbers; 
+            resultMessage = MutipleFuncMethod(5, 3);
+            Console.WriteLine($"Func without lambda = {resultMessage}");
+
+            //with lambda resultMessage = MutipleFuncMethod(5, 3);
```
Comment (CodeMazeBlog) [inline csharp-advanced-topics/FuncInCSharp/FuncInCSharp/Program.cs:15]: Remove the comments.

File: csharp-advanced-topics/ActionInCSharp/ActionInCSharp/Program.cs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+            Action<string> MessageView = ShowMessage;
+            MessageView("Code Maze in Action");
+            Console.WriteLine($"Action with single = {resultMessage}");
+
+            //Without lambda MultiplicationMethod(10, 20);
+            Action<int, int> MultiplicationMethod = MultiplyNumbers; 
+            MultiplicationMethod(10, 20);
+            Console.WriteLine($"Action without lambda value = {resultNumber}");
+
+            //With lambda MultiplicationMethodLambda(10, 20);
```
Comment (CodeMazeBlog) [inline csharp-advanced-topics/ActionInCSharp/ActionInCSharp/Program.cs:21]: Remove the comments.

**Review by CodeMazeBlog — COMMENTED** [review]
Small fixes needed. Also, now you've created a PR from your main branch. Next time you do, you'll probably drag these commits along. So please make sure to clean the fork after this article, and then use the branches for each new article and create PRs from these branches to our main.

File: csharp-advanced-topics/ActionInCSharp/ActionInCSharp/Program.cs (hunk 1 above)
Comment (namcogit) [reply csharp-advanced-topics/ActionInCSharp/ActionInCSharp/Program.cs:9]: I've created branch from my main branch named as "naman/ActionAndFunc" but cannot make commit and push it gives me.
image

Now as I want to commit the changes to this new branch and then make a PR to your main branch.

Comment (CodeMazeBlog) [reply csharp-advanced-topics/ActionInCSharp/ActionInCSharp/Program.cs:9]: You're trying to push directly to our repository. You need to use your own fork, and create a PR from that fork to our repo.

**Pull request closed without merging**